src/components/section/ohri-form-section.component.tsx

```diff
@@ -38,30 +47,85 @@ const OHRIFormSection = ({ fields, onFieldChange }) => {
           .filter((entry) => entry?.fieldComponent)
           .map((entry, index) => {
             const { fieldComponent: FieldComponent, fieldDescriptor, handler } = entry;
+
+            const prevValue = encounterContext.previousEncounter
+              ? handler?.getPreviousValue(fieldDescriptor, encounterContext.previousEncounter, fieldsFromEncounter)
+              : { value: 'no previous value' };
+
             if (FieldComponent) {
+              const previousValue = previousValues?.find((searchItem) => searchItem.field === fieldDescriptor.id);
               const qnFragment = (
                 <FieldComponent
                   question={fieldDescriptor}
                   onChange={onFieldChange}
                   key={index}
                   handler={handler}
                   useField={useField}
+                  previousValue={previousValue}
```

Comment:
I have a few questions regarding this approach...

- If we are picking the field's previous value from the previous encounter, what is the reason for subjecting the values to form validators again if these were already submitted, presumably having passed the validation before being submitted?
- What are the improvements you suggest in order to better this approach?
- Once this implementation is in, while migrating to React Form Hook, are we simply going to scrub it in favour of a new implementation? Because in that case I feel it wiser to wait until the migration in order to handle this any which way, no?

Reply:
> If we are picking the field's previous value from the previous encounter, what is the reason for subjecting the values to form validators again if these were already submitted, presumably having passed the validation before being submitted?

- **Existence of time-sensitive business logic**: Certain data may have expiration dates or validity periods, and revalidation ensures that the information remains accurate and up-to-date.
- **Revised business logic for new form iterations**: In cases where we have new form versions, the validation criteria or rules may also change. Revalidating previously submitted values ensures that they conform to the latest business logic and validation rules.
- **Revalidation considering changes in related values**: Business logic may depend on the relationships between different data points or values. If other related values have changed since the previous encounter, it may affect the validity or interpretation of the previously submitted values.

> Because in that case I feel it wiser to wait until the migration in order to handle this any which way, no?

I would say yes and no! Yes, because we ought to migrate anyway. No, because I believe there should be a simpler way of achieving the UI fixes without this expensive approach. Consider the transition to Form Hook as a potential breaking change that will require a major release.

Hear me out: As much as this doesn't seem like a breaking change, I would say it's a major change that may introduce unknown side effects. This doesn't seem like a major revamp that addresses performance-related issues or restructuring components to make maintenance simpler, but rather a UX bump.

Question: Is it possible to address the UI issues without resorting to this approach? *(I'm not conservative about the legacy patterns but rather a little worried about how we are approaching things here.)*

While it makes sense for the section (parent) to manage the rendering of its questions (children), I don't think it's a good idea for it to directly manage the states of their previous values. In my pov, I would suggest addressing the UI issues tangential to previous values by using robust CSS selectors instead of the DOM shift. We can then set up an epic to address most of these technical debts, such as decoupling validation and business logic processing from the [ohri-form.component.tsx](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/ohri-form.component.tsx).

Reply:
While I do agree that this is a large change, I still don't believe there is a feasible way to achieve this with only CSS, unless of course you are open to discourse and finding a solution we can both agree on?

Reply:
@samuelmale as much as we appreciate all the valuable comments/inputs given to date. We need this work merged for because we need it for our next release. Regarding the approaches you are proposing I would suggest you propose them in form of new tickets on the [React Form Engine NEXT](https://openmrs.atlassian.net/browse/O3-2193)  epic on Jira. Unless there is an obvious bug lets merge this PR please. Am also open to have a call to discuss this. Thank you